### PR TITLE
Sharing integration test 20645, added sabre dav and propfind check.

### PR DESCRIPTION
**Pull request by SergioBertolinSG**
Includes Morris's test and the webdav required functionality using sabre dav.

### CODE REVIEW

**Comment by SergioBertolinSG** [comment]
@MorrisJobke @PVince81 Please review.

File: build/integration/features/sharing-v1.feature [hunk 1]
```diff
@@ -363,6 +363,22 @@ Feature: sharing
     Then the OCS status code should be "404" 
     And the HTTP status code should be "200"
 
+  Scenario: Share of folder and sub-folder to same user - core#20645
+    Given As an "admin"
+    And user "user0" exists
+    And user "user1" exists
+    And group "group0" exists
+    And user "user1" belongs to group "group0"
+    And file "/PARENT" from user "user0" is shared with user "user1"
```
Comment (MorrisJobke) [inline build/integration/features/sharing-v1.feature]: I would name it `file ABC of user DEF is shared with user GHI`

**Comment by MorrisJobke** [comment]
Code looks good :+1:

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**